specs/governance/timelock-guard.md

```diff
@@ -0,0 +1,136 @@
+# Governor
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Definitions](#definitions)
+  - [`quorum(safe)`](#quorumsafe)
+  - [Scheduled Transaction](#scheduled-transaction)
+  - [`scheduling_time(safe, tx)`](#scheduling_timesafe-tx)
+  - [`delay_period(safe)`](#delay_periodsafe)
+  - [Rejected Transaction](#rejected-transaction)
+  - [`rejecting_owners(safe, tx)`](#rejecting_ownerssafe-tx)
+  - [`cancellation_threshold(safe)`](#cancellation_thresholdsafe)
+  - [Nested Safe Setup](#nested-safe-setup)
+  - [Nested Cancellation](#nested-cancellation)
+- [Assumptions](#assumptions)
+- [Invariants](#invariants)
+- [Function Specification](#function-specification)
+  - [constructor](#constructor)
+  - [enable(delay_period)](#enabledelay_period)
+  - [disable](#disable)
+  - [cancellationThreshold(safe)](#cancellationthresholdsafe)
+  - [delayPeriod(safe)](#delayperiodsafe)
+  - [setDelayPeriod(delay_period)](#setdelayperioddelay_period)
+  - [scheduleTransaction](#scheduletransaction)
+  - [checkTransaction](#checktransaction)
+  - [rejectTransaction](#rejecttransaction)
+  - [cancelTransaction](#canceltransaction)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The `TimelockGuard` implements a mandatory delay on Gnosis Safe contracts as a Safe Guard. Safe Guards are external contracts that Safes call into before and after the execution of a transaction to enquire if the execution fulfils conditions implemented in the Guard, and revert if they don't. The `TimelockGuard` 
+
+## Definitions
+
+### `quorum(safe)`
+The number of owners in a safe that must signal aproval of a transaction for it to be executed by the safe.
+
+### Scheduled Transaction
+A specific transaction that has been stored in the Timelock, for execution from a given safe.
+
+### `scheduling_time(safe, tx)`
+The time in seconds of the block in which a transaction was scheduled.
+
+### `delay_period(safe)`
+Given a scheduled transaction, the minimum time in seconds after `scheduling_time` that execution is allowed.
+
+### Rejected Transaction
+Given a transaction scheduled to execute from given safe, owners for the safe reject the transaction to signal they think the transaction should not be executed after the `delay_period`.
+
+### `rejecting_owners(safe, tx)`
+Given a transaction scheduled to execute from given safe, the owners for the safe that rejected the transaction.
+
+### `cancellation_threshold(safe)`
+The number of owners that must reject a given transaction for it not to be executed after the `delay_period`. It is equal to `blocking_minority`, where `blocking_minority = min(quorum(safe), total_owners(safe) - quorum + 1)`.
```

Comment:
We should rename `blocking_minority` to `blocking_threshold` since it may not always be a minority